libp2p.go

```diff
@@ -0,0 +1,181 @@
+package libp2p
+
+import (
+	"context"
+	"crypto/rand"
+
+	crypto "github.com/libp2p/go-libp2p-crypto"
+	host "github.com/libp2p/go-libp2p-host"
+	pnet "github.com/libp2p/go-libp2p-interface-pnet"
+	metrics "github.com/libp2p/go-libp2p-metrics"
+	peer "github.com/libp2p/go-libp2p-peer"
+	pstore "github.com/libp2p/go-libp2p-peerstore"
+	swarm "github.com/libp2p/go-libp2p-swarm"
+	transport "github.com/libp2p/go-libp2p-transport"
+	bhost "github.com/libp2p/go-libp2p/p2p/host/basic"
+	mux "github.com/libp2p/go-stream-muxer"
+	ma "github.com/multiformats/go-multiaddr"
+	mplex "github.com/whyrusleeping/go-smux-multiplex"
+	msmux "github.com/whyrusleeping/go-smux-multistream"
+	yamux "github.com/whyrusleeping/go-smux-yamux"
+)
+
+// Config describes a set of settings for a libp2p node
+type Config struct {
+	Transports   []transport.Transport
+	Muxer        mux.Transport
+	ListenAddrs  []ma.Multiaddr
+	PeerKey      crypto.PrivKey
+	Peerstore    pstore.Peerstore
+	Protector    pnet.Protector
+	Reporter     metrics.Reporter
+	DisableSecio bool
+}
+
+type Option func(cfg *Config) error
+
+func Transports(tpts []transport.Transport) Option {
```

Comment:
The function signatures are inconsistent, would be nice if they all either take an array or varargs, as currently two of them (Transports and ListenAddrs)  take an array but ListenAddrStrings takes varargs.

Reply:
I may have been a little tired writing this... >.>